sklearn/preprocessing/tests/test_data.py

```diff
@@ -191,19 +210,269 @@ def test_scaler_2d_arrays():
     scaler = StandardScaler()
     X_scaled = scaler.fit(X).transform(X, copy=True)
     assert_false(np.any(np.isnan(X_scaled)))
-    assert_array_almost_equal(X_scaled.mean(axis=0), 5 * [0.0])
+    assert_array_almost_equal(X_scaled.mean(axis=0), n_features * [0.0])
     assert_array_almost_equal(X_scaled.std(axis=0), [0., 1., 1., 1., 1.])
     # Check that X has not been copied
     assert_true(X_scaled is not X)
 
 
+def test_handle_zeros_in_scale():
+    s1 = np.array([0, 1, 2, 3])
+    s2 = _handle_zeros_in_scale(s1, copy=True)
+
+    assert_array_equal(s1, np.array([0, 1, 2, 3]))
+    assert_array_equal(s2, np.array([1, 1, 2, 3]))
+    assert_false(s1[0] == s2[0])
```

Comment:
I think it is but only if the two tests above passed. So maybe we can keep the line but move it up.